pyirf/interpolators/base_interpolators.py

```diff
@@ -0,0 +1,241 @@
+                f" points needed to span a volume in {self.grid_dim} dimensions."
+            )
+
+        # Build triangulation to check if target is inside of the grid for
+        # more then 1 dimension
+        if self.grid_dim > 1:
+            try:
+                self.Triangulation = Delaunay(self.grid_points)
+            except QhullError:
+                raise
+
+    @abstractmethod
+    def interpolate(self, target_point, **kwargs):
+        """Overridable function for the actual interpolation code"""
+        pass
+
+    def _target_in_grid(self, target_point):
+        """Check wether target_point lies within grids convex hull"""
+        if self.grid_dim == 1:
+            return (target_point >= self.grid_points.min()) and (
+                target_point <= self.grid_points.max()
+            )
+        else:
+            # Delaunay.find_simplex() returns -1 for points outside the grids convex hull
+            simplex_ind = self.Triangulation.find_simplex(target_point)
+            return simplex_ind >= 0
+
+    def __call__(self, target_point, extrapolator=None, **kwargs):
+        """Providing a common __call__ interface sanity checking the target point
+
+        Parameters
+        ----------
+        target_point: np.ndarray
+            Target for inter-/extrapolation
+        extrapolator: callable, optional
+            Fall-Back extrapolator called when target_point is outside grid.
+            Defaults to None, meaning no extrapolation is done.
+
+        Raises
+        ------
+        TypeError:
+            When target_point is not an np.ndarray
+        ValueError:
+            When more then one target_point is given
+        ValueError:
+            When target_point and grid_points have miss-matching dimensions
+        ValueError:
+            When target_point is outside of the grids convex hull but extrapolator is None
+
+        Returns
+        -------
+        Interpolated or, if necessary extrapolated, result.
+        """
+        if not isinstance(target_point, np.ndarray):
+            raise TypeError("Target point is not a numpy array.")
+
+        if target_point.ndim == 1:
+            target_point = target_point.reshape(1, *target_point.shape)
+        elif target_point.shape[0] != 1:
+            raise ValueError("Only one target_point per call supported.")
+
+        if target_point.shape[1] != self.grid_dim:
+            raise ValueError(
+                "Missmatch between target-point and grid dimension."
+                f" Grid has dimension {self.grid_dim}, target has dimension"
+                f" {target_point.shape[1]}."
+            )
+
+        if self._target_in_grid(target_point):
+            return self.interpolate(target_point, **kwargs)
+        elif extrapolator is not None:
+            print(f"Trying to extrapolate for point {target_point}.")
+            try:
+                return extrapolator(target_point, **kwargs)
+            except:
```

Comment:
Don't use bare excepts, as these also catch `SystemExit` and `KeyboardInterrupt`. 

Using `except Exception` will catch all other exceptions.

However, here you anyway only reraise, so the try except block should just completely be removed